src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs

```diff
@@ -0,0 +1,536 @@
+        {
+            Debug.Assert(value != null);
+
+            // The 32-bit modulus pow method for a big integer
+            // raised by a 32-bit integer...
+
+            uint temp = Remainder(value, modulus)[0];
+
+            uint result = PowCore(power, modulus, temp, 1);
+
+            return new uint[] { result };
+        }
+
+        public static uint[] Pow(uint value, uint[] power, uint modulus)
+        {
+            Debug.Assert(power != null);
+
+            // The 32-bit modulus pow method for a 32-bit integer
+            // raised by a big integer...
+
+            uint result = PowCore(power, modulus, value, 1);
+
+            return new uint[] { result };
+        }
+
+        public static uint[] Pow(uint[] value, uint[] power, uint modulus)
+        {
+            Debug.Assert(value != null);
+            Debug.Assert(power != null);
+
+            // The 32-bit modulus pow method for a big integer
+            // raised by a big integer...
+
+            uint temp = Remainder(value, modulus)[0];
+
+            uint result = PowCore(power, modulus, temp, 1);
+
+            return new uint[] { result };
+        }
+
+        private static uint PowCore(uint[] power, uint modulus,
+                                    ulong temp, ulong result)
+        {
+            // The 32-bit modulus pow algorithm for all but
+            // the last power limb using square-and-multiply.
+
+            for (int i = 0; i < power.Length - 1; i++)
+            {
+                uint p = power[i];
+                for (int j = 0; j < 32; j++)
+                {
+                    if ((p & 1) == 1)
+                        result = (result * temp) % modulus;
+                    temp = (temp * temp) % modulus;
+                    p = p >> 1;
+                }
+            }
+
+            return PowCore(power[power.Length - 1], modulus, temp, result);
+        }
+
+        private static uint PowCore(uint power, uint modulus,
+                                    ulong temp, ulong result)
+        {
+            // The 32-bit modulus pow algorithm for the last or
+            // the only power limb using square-and-multiply.
+
+            while (power != 0)
+            {
+                if ((power & 1) == 1)
+                    result = (result * temp) % modulus;
+                temp = (temp * temp) % modulus;
+                power = power >> 1;
+            }
+
+            return (uint)(result % modulus);
+        }
+
+        public static uint[] Pow(uint value, uint power, uint[] modulus)
+        {
+            Debug.Assert(modulus != null);
+
+            // The big modulus pow method for a 32-bit integer
+            // raised by a 32-bit integer...
+
+            BitsBuffer temp = new BitsBuffer(modulus.Length * 2, value);
+            BitsBuffer result = new BitsBuffer(modulus.Length * 2, 1);
+            FastReducer reducer = new FastReducer(modulus);
```

Comment:
Sounds good, thanks.  It'd be good to also try running some perf tests in parallel, both with and without the server GC enabled, to see how things are affected.